packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -677,21 +681,31 @@ class _DraggableScrollableSheetState extends State<DraggableScrollableSheet> {
 
   @override
   Widget build(BuildContext context) {
-    return ValueListenableBuilder<double>(
+    final Widget child = ValueListenableBuilder<double>(
       valueListenable: _extent._currentSize,
-      builder: (BuildContext context, double currentSize, Widget? child) => LayoutBuilder(
-        builder: (BuildContext context, BoxConstraints constraints) {
-          _extent.availablePixels = widget.maxChildSize * constraints.biggest.height;
-          final Widget sheet = FractionallySizedBox(
-            heightFactor: currentSize,
-            alignment: Alignment.bottomCenter,
-            child: child,
-          );
-          return widget.expand ? SizedBox.expand(child: sheet) : sheet;
-        },
-      ),
+      builder: (BuildContext context, double currentSize, Widget? child) =>
+        FractionallySizedBox(
+          heightFactor: currentSize,
+          alignment: Alignment.bottomCenter,
+          child: child,
+        ),
       child: widget.builder(context, _scrollController),
     );
+
+    final Widget wrappedChild = widget.expand ?
+      SizedBox.expand(child: child) : child;
+
+    return LayoutBuilder(
```

Comment:
It might be more idiomatic to just build the `SizedBox` inline in the `LayoutBuilder`, but I feel liek we shouldn't really even need a `LayoutBuilder` here since the widget subtree itself doesn't depend on the layout at all, but I don't know of a better way to add a layout callback like this. (All we're trying to do is call `applyNewOuterDimensions` to update the scroll snap animations on constraint changes.)